frontend/simple/js/idb-flume-log.js

```diff
@@ -0,0 +1,192 @@
+import localforage from 'localforage'
+import { toHash, makeEntry } from '../../../shared/functions'
+import Append from 'append-batch'
+import Obv from 'obv'
+const storeName = 'EventLogStorage'
+const name = 'Group Income'
+
+// Generate a log
+export default async function log (vuex) {
+  // Initialize localforage to Index
+  // https://www.html5rocks.com/en/tutorials/offline/quota-research/#toc-overview
+  localforage.config({
+    driver: localforage.INDEXEDDB,
+    name,
+    storeName
+  })
+  /* eslint-disable no-unused-vars */
+  let current
+  // since is the Observable object
+  // https://github.com/dominictarr/obv
+  let since = Obv()
+  let store = vuex
+
+  // Load current state of the log
+  try {
+    current = await localforage.getItem('current')
+  } catch (ex) {
+    current = null
+  }
+  if (store) {
+    store.commit('UPDATELOG', current)
+  }
+  since.set(current)
+
+  function get (opts, cb) {
+    let hash
+    if (typeof opts === 'string') {
+      hash = opts
+    } else {
+      hash = opts.hash
+    }
+    localforage.getItem(hash, (err, entry) => {
+      if (err) {
+        return cb(err)
+      }
+      if (opts.parentHash) {
+        return cb(null, entry.parentHash)
+      } else {
+        return cb(null, entry.data)
+      }
+    })
+  }
+
+  let batcher = function (batch, cb) {
+    // DB must be canonical source of current log may lose sync with two windows open
```

Comment:
Sorry, to clarify: I considered that if you have two windows on the same application then they have two different memory spaces tracking the current position seperately. This may cause conflict on appends.